### PR TITLE
fs.existsSync deprecated, replace with exists-sync

### PR DESCRIPTION
**Pull request by jasonmit**
https://github.com/ember-cli/ember-cli/issues/4315

~~I took a first pass at this, noticed the tests should likely be using tests/helpers/file-utils where possible for exists. Want me to address this as well @stefanpenner~~ 

I went ahead and updated the tests to use `file-utils.exists`

I will squash once thumbs up

### CODE REVIEW

File: tests/helpers/acceptance.js [hunk 1]
```diff
@@ -26,10 +27,10 @@ function downloaded(item) {
   var exists = false;
   switch (item) {
     case 'node_modules':
-      exists = fs.existsSync(path.join(root, '.node_modules-tmp'));
+      exists = exists(path.join(root, '.node_modules-tmp'));
```
Comment (trabus) [inline tests/helpers/acceptance.js]: Will need to change these so you're not overwriting exists. :)

File: tests/helpers/acceptance.js [hunk 2]
```diff
@@ -26,10 +27,10 @@ function downloaded(item) {
...
       break;
     case 'bower_components':
-      exists = fs.existsSync(path.join(root, '.bower_components-tmp'));
+      exists = exists(path.join(root, '.bower_components-tmp'));
```
Comment (trabus) [inline tests/helpers/acceptance.js]: Same here.

File: tests/helpers/acceptance.js (hunk 1 above)
Comment (jasonmit) [inline tests/helpers/acceptance.js]: :+1:

**Comment by trabus** [comment]
I'm wondering if maybe we should go with `existsSync` for all of it, to prevent any ambiguity about whether it's the async exists or not.

**Comment by jasonmit** [comment]
> I'm wondering if maybe we should go with existsSync for all of it, to prevent any ambiguity about whether it's the async exists or not.

Yup, I agree.  I'll update accordingly.

File: tests/acceptance/brocfile-smoke-test-slow.js [hunk 3]
```diff
@@ -392,7 +393,7 @@ describe('Acceptance: brocfile-smoke-test', function() {
 
     return runCommand(path.join('.', 'node_modules', 'ember-cli', 'bin', 'ember'), 'build', '--silent')
       .then(function() {
-        var exists = fs.existsSync(path.join('.', 'dist', 'assets', appName + '.css'));
+        var exists = exists(path.join('.', 'dist', 'assets', appName + '.css'));
```
Comment (trabus) [inline tests/acceptance/brocfile-smoke-test-slow.js]: Looks like another one here, but it won't matter when you've changed everything to use `existsSync` instead.

**Comment by trabus** [comment]
LGTM :+1: 

@stefanpenner

**Comment by jasonmit** [comment]
Squashed, thanks @trabus for the review

**Comment by stefanpenner** [comment]
Looks good. That file-utile relative import should likely just point to the actual existsSync no?

**Comment by jasonmit** [comment]
@stefanpenner yes and should already be doing that in this PR

File: tests/acceptance/brocfile-smoke-test-slow.js [hunk 4]
```diff
@@ -12,6 +12,7 @@ var runCommand          = require('../helpers/run-command');
 var acceptance          = require('../helpers/acceptance');
 var copyFixtureFiles    = require('../helpers/copy-fixture-files');
 var assertDirEmpty      = require('../helpers/assert-dir-empty');
+var existsSync          = require('../helpers/file-utils').existsSync;
```
Comment (stefanpenner) [inline tests/acceptance/brocfile-smoke-test-slow.js]: @jasonmit what about these?

Comment (jasonmit) [inline tests/acceptance/brocfile-smoke-test-slow.js]: file-utils uses exists-sync now https://github.com/jasonmit/ember-cli/blob/exist-sync/tests/helpers/file-utils.js#L4

Perhaps I'm misunderstanding

Comment (stefanpenner) [inline tests/acceptance/brocfile-smoke-test-slow.js]: yes, but why bother with the intermediate file at all?

Comment (jasonmit) [inline tests/acceptance/brocfile-smoke-test-slow.js]: Okay, will change this shortly

**Comment by jasonmit** [comment]
@stefanpenner updated to remove intermediate file

**Comment by stefanpenner** [comment]
Awesome thanks

**Comment by trabus** [comment]
@jasonmit Thanks for taking care of this!

**Pull request merged**